### ISSUE TITLE
Question about: Why depreciate KopiLua?

### ISSUE THREAD

**Issue by chansey97** [Mervill/Unity3D-NLua#11]
Can you tell me some reason about: why depreciate KopiLua?
In my view, even though Unity 5 allowed Indie licnence use native dlls, support KopiLua is not a bad idea.

**Comment by Mervill** [comment]
A few reasons;
1. My focus at the moment is primarily PC development, since thats where I see the most potential for LUA as a useful extension.
2. KopiLua itself is not particularly slow, but performance in Unity3D (which, among other things, has a vastly outdated GC) was not looking very good. These issues can obviously be solved, but there are other issues I'd like to address first.
3. KopiLua currently targets lua 5.1, while KeraLua targets lua 5.2, although it's only a minor revision, lua 5.2 depreciated the functions `getfenv` and `setfenv` in favour of `_env` as a magic variable. Creating a framework that can deal with both systems is non-trivial. Obviously it _is_ a solvable problem, but not one I'm interested in pursuing at the moment.

tl;dr, I could support KopiLua, but that's not my priority right now. At the very least I'll probably keep my KopiLua fork updated so that it compiles in Unity, but I don't have time to support it at the same level as this project.

If someone else would like to take ownership of KopiLua compatibility, I'd be happy to accept pull requests =D
